controller.rst

```diff
@@ -244,41 +237,136 @@ The Symfony templating system and Twig are explained more in the
 
 .. _controller-accessing-services:
 
-Accessing other Services
-~~~~~~~~~~~~~~~~~~~~~~~~
+Fetching Services as Arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Symfony comes packed with a lot of useful objects, called *services*. These
-are used for rendering templates, sending emails, querying the database and
-any other "work" you can think of. When you install a new bundle, it probably
-brings in even *more* services.
+.. versionadded:: 3.3
+    The ability to type-hint an argument in order to receive a service was added
+    in Symfony 3.3.
 
-When extending the base ``Controller`` class, you can access any Symfony service
-via the :method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::get`
-method. Here are several common services you might need::
+Symfony comes *packed* with a lot of useful objects, called :doc:`services </service_container>`.
+These are used for rendering templates, sending emails, querying the database and
+any other "work" you can think of.
 
-    $templating = $this->get('templating');
+If you need a service, just type-hint an argument with its class (or interface) name.
+Symfony will automatically pass you the service you need::
 
-    $router = $this->get('router');
+    use Psr\Log\LoggerInterface
+    // ...
 
-    $mailer = $this->get('mailer');
+    /**
+     * @Route("/lucky/number/{max}")
+     */
+    public function numberAction($max, LoggerInterface $logger)
+    {
+        $logger->info('We are logging!');
+        // ...
+    }
 
-What other services exist? To list all services, use the ``debug:container``
-console command:
+Awesome!
+
+What other services exist? Each page of the documentation will reveal more and more
+services you can use. To list *all* services, use the ``debug:container`` console
+command:
 
 .. code-block:: terminal
 
     $ php bin/console debug:container
 
-For more information, see the :doc:`/service_container` article.
+If need to control the *exact* value of an argument, you can override your controller's
+service config:
 
-.. tip::
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            # ...
+
+            # explicitly configure the service
+            AppBundle\Controller\LuckyController:
+                public: true
+                tags:
+                    # add multiple tags to controller multiple args
+                    - name: controller.service_arguments
+                      action: numberAction
+                      argument: logger
+                      # pass this specific service id
+                      id: monolog.logger.doctrine
```

Comment:
I also like bindings... we'll see how everything works out for in reality in Symfony 3.3 :). For the most part, bindings and arguments accomplish the same task (in a slightly different way). But it's interesting that bindings could also work to help with controller args - I hadn't really thought about that... seems like one use-case that makes bindings actually a bit better.